### ISSUE TITLE
Add `flatTransform` to `OptionT`

### ISSUE THREAD

**Issue by amilkov3** [typelevel/cats#2309]
```scala
def flatTransform[F[_]: FlatMap, A, B](
  t: OptionT[F, A], 
  f: Option[A] => F[Option[B]]
): OptionT[F, B] = OptionT(t.value.flatMap(f))
```

**Comment by abhishek7** [comment]
@amilkov3 was giving this a try, but a bit confused with `OptionT(t.value.flatMap(f))` on the last line. Isn't `flatMap` not a member of `F[Option[A]]`?

**Comment by amilkov3** [comment]
@abhishek7 it is if you give `F` a `FlatMap` constraint

**Comment by kubukoz** [comment]
Since only `flatMap` is needed, I'd leave a constraint just for `FlatMap`.

**Comment by abhishek7** [comment]
@amilkov3 I like the alternative you provided! If it's alright with you, I'll open up a quick PR with that implementation along with a test? Let me know if you, @kubukoz, or anyone else has any other suggestions.

**Comment by kubukoz** [comment]
Closed by #2314
